source/source.go

```diff
@@ -26,6 +26,7 @@ import (
 
 var (
 	_ Iterator = (*logrepl.CDCIterator)(nil)
```

Comment:
After we have this new iterator we can remove the interface check for `CDCIterator`, we don't care what it implements anymore, `Source` doesn't even have to know it exists.